### PR TITLE
Version 1.1.0

### PR DESCRIPTION
**Pull request by barthopster**
***Context***

To make it a better fit for the current Nedap applications, Segnify has been majorly refactored. The 'content scroll view' has been replaced by a `UIPageViewController` implementation, the level of customization has been increased and many general improvements have been made.

***Changes***
#### Added

- Added `PageViewController`, which replaces the usage of any content `UIScrollView`. `PageViewController` contains an implementation of `Segnify` and an instance of `UIPageViewController`, which are connected to each other.
- Added `PageViewControllerDataSourceProtocol`. The delegate which implements this protocol is the data source for the `PageViewController` instance being used.
- Added `PageViewControllerProtocol`, which can be used for customizing the UI of the `PageViewController` instance. Currently, the height of the `Segnify` instance inside the `PageViewController` instance can be configured.
- Added `SegnifyDataSourceProtocol`. The delegate which implements this protocol is the data source for the `Segnify` instance being used.
- Added `SegnifyEventsProtocol`. The delegate which implements this protocol will be notified about `Segment` selection changes of `Segnify`.
- Added `DefaultDelegates`, which offers a default implementation of almost all protocols.

#### Removed
- Removed `BaseView`.

#### Updated
- Renamed all `*Configuration` protocols to `*Protocol`.
- Updated the `Example` applications accordingly.

***Check-list***
- [ ] Acceptance criteria in issue satisfied or Bug fixed
- [ ] Documentation present (relevant information is properly documented)
- [ ] Tested on devices
- [ ] CHANGELOG updated

### CODE REVIEW

File: Example/Carthage/Cartfile [hunk 1]
```diff
@@ -1 +1 @@
-github "nedap/Segnify" ~> 1.0.2
\ No newline at end of file
+github "nedap/Segnify" ~> 1.1.0
```
[the remark this answers is not in the document]
Comment (martias) [reply Example/Carthage/Cartfile]: Haha, similar for CocoaPods to test the last commits ;)
```
pod 'Segnify', :git => 'https://github.com/nedap/Segnify.git', :commit => 'e3bf1f7012c7d29ce3802b1269238f8a32f9d84e'
```

File: Example/Carthage/Segnified/View Controllers/MainViewController.swift [hunk 2]
```diff
@@ -10,123 +10,4 @@ import UIKit
...
-// MARK: - Segnify delegate
-
-extension MainViewController {
-    
-    func segnify(_ segnify: Segnify, didSelect segment: Segment, with index: Int) {
-        // Scroll the content scroll view to the corresponding 'page'.
-        contentScrollView.setContentOffset(CGPoint(x: (CGFloat(index) * contentScrollView.frame.width), y: 0.0), animated: true)
-    }
-}
+public class MainViewController: PageViewController {}
```
[the remark this answers is not in the document]
Comment (barthopster) [reply Example/Carthage/Segnified/View Controllers/MainViewController.swift]: Solved that in the [feature/4_improved_dev_test_flow](https://github.com/nedap/Segnify/tree/feature/4_improved_dev_test_flow) branch.

File: Example/CocoaPods/Segnified/View Controllers/MainViewController.swift [hunk 3]
```diff
@@ -10,123 +10,4 @@ import UIKit
...
-// MARK: - Segnify delegate
-
-extension MainViewController {
-    
-    func segnify(_ segnify: Segnify, didSelect segment: Segment, with index: Int) {
-        // Scroll the content scroll view to the corresponding 'page'.
-        contentScrollView.setContentOffset(CGPoint(x: (CGFloat(index) * contentScrollView.frame.width), y: 0.0), animated: true)
-    }
-}
+public class MainViewController: PageViewController {}
```
[the remark this answers is not in the document]
Comment (barthopster) [reply Example/CocoaPods/Segnified/View Controllers/MainViewController.swift]: Solved that in the [feature/4_improved_dev_test_flow](https://github.com/nedap/Segnify/tree/feature/4_improved_dev_test_flow) branch.

File: Segnify/View Controllers/PageViewController.swift [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+    
+    // MARK: - Setup
+    
+    public func setup(dataSource: PageViewControllerDataSourceProtocol? = DefaultDelegates.shared,
```
[the remark this answers is not in the document]
Comment (barthopster) [reply Segnify/View Controllers/PageViewController.swift]: My initial thought was to make it public, in order to give the end user multiple possibilities for setting / changes the data source and delegate objects. I made this method private, the end user can still supply a custom data source and custom delegate through the `convenience init` and the `dataSource` and `delegate` properties.
The idea was based on Apple's UITableView, where you can change its `dataSource` and `delegate` whenever you want as well.

[the remark this answers is not in the document]
Comment (barthopster) [reply Segnify/View Controllers/PageViewController.swift]: You're totally right, nice idea. I created the `SegnifyContentElement` tuple, which represents a `Segment` instance and a `UIViewController` instance.

File: Segnify/View Controllers/PageViewController.swift [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            segnify.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            segnify.topAnchor.constraint(equalTo: topLayoutGuide.bottomAnchor),
+            segnifyHeightConstraint!
+            ], for: segnify)
+        
+        // Populate.
+        segnify.populate()
+        
+        // Add the page view controller.
+        addChild(pageViewController)
```
[the remark this answers is not in the document]
Comment (barthopster) [reply Segnify/View Controllers/PageViewController.swift]: True! Done.

**Comment by barthopster** [comment]
@dregatos & @martias, thank you guys for the feedback! Finally I destroyed all the bugs.

I tested it on iOS 9, 10 and 12, on iPhone 6s, 8 and XS. Everything works as intended now. On iOS 10, the UI was glitchy indeed, but that has been fixed.

Please take a look, as I made quite some changes. I can do a show-and-tell for some detailed explanation 👍.

**Comment by barthopster** [comment]
Please refer to #5.

**Pull request closed without merging**